.github/workflows/linter.yml

```diff
@@ -0,0 +1,23 @@
+name: Java Linter
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - name: Install dependencies
+        run: npm install
+      - name: Run checkstyle
+        run: npm run linter
```

Comment:
does doing all this and the build make sense?

Reply:
I'm not really sure. I thought they had to be separate to show up as separate checks in GitHub

Reply:
so that's it, I don't think we need separate yml files. Let's just add       

```
- name: Run checkstyle
   run: npm run linter 
```

to the build.yml

This will be quicker because I'm not pulling in these actions multiple times and installing dependencies.